ios/Exponent/Versioned/Core/Api/EXNotifications.m

```diff
@@ -141,13 +217,14 @@ - (instancetype)initWithExperienceId:(NSString *)experienceId kernelServiceDeleg
                  cancelAllScheduledNotificationsAsyncWithResolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(__unused RCTPromiseRejectBlock)reject)
 {
-  [EXUtil performSynchronouslyOnMainThread:^{
-    for (UILocalNotification *notification in [RCTSharedApplication() scheduledLocalNotifications]) {
-      if ([notification.userInfo[@"experienceId"] isEqualToString:self.experienceId]) {
-        [RCTSharedApplication() cancelLocalNotification:notification];
+  [[UNUserNotificationCenter currentNotificationCenter] getPendingNotificationRequestsWithCompletionHandler:
```

Comment:
Doesn't it need to run on the main thread?

Reply:
"You may use the shared user notification center object simultaneously from any of your app's threads. The object processes requests serially in the order they were initiated."
https://developer.apple.com/documentation/usernotifications/unusernotificationcenter?language=objc

Reply:
In iOS 11 the guidance was:

>Thread-Safety Information
>
>You may use the shared user notification center object from any thread of your app. However, you should use this object from only one thread at a time. Do not try to use it from multiple threads simultaneously. Always assign your delegate object to the shared user notification center's delegate property before using the object. It is a programmer error to assign a delegate after calling methods that might return information to that delegate.

So for backwards compatibility I think we may need to use a consistent queue everywhere. But unlike with UIApplication, we don't need to use the main queue.

Reply:
I just didn't notice that. I think that we can wrap unusernotificationcenter (only used methods) in a class and each method  will be pushing block to our queue.